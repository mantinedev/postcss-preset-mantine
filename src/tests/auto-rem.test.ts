import { testTransform } from './utils';

const baseInput = `
.demo {
  font-size: 16px;
  height: 40px;
  padding: 10px 50px 20px 30px;
  background: red;
  border: 1px solid black;
  box-shadow: 0 0 10px 5px rgba(0, 0, 0, 0.5);
}
`;

const remFunctionInput = `
.demo {
  font-size: rem(16px);
  border: rem(1px) solid red;
  padding: 16px;
}
`;

const mediaInput = `
.demo {
  @media (min-width: 320px) {
    font-size: 32px;
  }
}
`;

describe('auto-rem', () => {
  it('it transforms px to rem values correctly', async () => {
    const res = await testTransform(baseInput, { autoRem: true });
    expect(res.css).toMatchSnapshot();
  });

  it('it transforms input with rem function correctly', async () => {
    const res = await testTransform(remFunctionInput, { autoRem: true });
    expect(res.css).toMatchSnapshot();
  });

  it('it transforms media query correctly', async () => {
    const res = await testTransform(mediaInput, { autoRem: true });
    expect(res.css).toMatchSnapshot();
  });
});
