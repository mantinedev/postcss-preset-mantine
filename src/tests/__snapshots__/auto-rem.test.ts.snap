// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auto-rem it transforms input with rem function correctly 1`] = `
"
.demo {
  font-size: calc(1rem * var(--mantine-scale));
  border: calc(0.0625rem * var(--mantine-scale)) solid red;
  padding: calc(1rem * var(--mantine-scale));
}
"
`;

exports[`auto-rem it transforms media query correctly 1`] = `
"
@media (min-width: 320px) {
.demo {
    font-size: calc(2rem * var(--mantine-scale))
}
  }
"
`;

exports[`auto-rem it transforms px to rem values correctly 1`] = `
"
.demo {
  font-size: calc(1rem * var(--mantine-scale));
  height: calc(2.5rem * var(--mantine-scale));
  padding: calc(0.625rem * var(--mantine-scale)) calc(3.125rem * var(--mantine-scale)) calc(1.25rem * var(--mantine-scale)) calc(1.875rem * var(--mantine-scale));
  background: red;
  border: calc(0.0625rem * var(--mantine-scale)) solid black;
  box-shadow: 0 0 calc(0.625rem * var(--mantine-scale)) calc(0.3125rem * var(--mantine-scale)) rgba(0, 0, 0, 0.5);
}
"
`;
